sympy/sets/fancysets.py

```diff
@@ -404,6 +413,62 @@ def _intersect(self, other):
                             self.base_set.intersect(
                                 solveset_real(im, n_)))
 
+        elif isinstance(other, Interval):
+            from sympy.solvers.solveset import (invert_real, invert_complex,
+                                                solveset)
+
+            f = self.lamda.expr
+            n = self.lamda.variables[0]
+            base_set = self.base_set
+            inf, sup = other.inf, other.sup
+            new_inf, new_sup = None, None
+
+            if f.is_real:
+                inverter = invert_real
+            else:
+                inverter = invert_complex
+
+            g1, h1 = inverter(f, inf, n)
```

Comment:
Use `other.inf` and `other.sup` instead of assigning names `inf` and `sup`. The code is more explicit that way.